sips/sip-021/sip-021-trustless-two-way-peg-for-bitcoin.md

```diff
@@ -0,0 +1,1385 @@
+
+In doing so, the system creates a new miner rule: a miner can mine off of any
+descendant of the highest frozen block-commit (i.e. any unblessed
+block-commit), and it can mine off of any descendant of a blessed block-commit.
+Miners freely and individually choose which member of this set of block-commits
+to mine off of.  A miner cannot mine off of any other block-commits.
+
+### Finalization
+
+Once a block-commit receives 100 Bitcoin confirmations, it can no longer be
+blessed.
+
+When Stackers fulfill a peg-out request, they invalidate all Stacks forks in
+which the peg-out would result in an inconsistent chain state (i.e. someone's
+sBTC balance becomes negative).  These forks will never be treated as canonical
+as long as they contain such inconsistencies.  As such, the act of Stackers
+fulfilling a peg-out request that materializes in Stacks block B is declaring
+that any Stacks fork that descends from B might be canonical (i.e. the
+canonical fork must include B).
+
+Consequently, the act of processing a peg-out is used to keep the chain alive
+even if someone manages to mine 99 block-commits without disclosing their
+blocks.  As long as there exists at least one valid publicly-disclosed block,
+then Stackers can use it to authenticate and process peg-outs and in doing so,
+require that block to be part of the canonical chain.  This enables miners and
+Stackers to prevent a hidden-block miner from halting the chain while spending
+far fewer resources than the attacker.  It also prevents a 51% attacker from
+denying sBTC transfers and peg-outs: if any sBTC transfer materializes in the
+honest fork, and Stackers process at least one peg-out on that fork, then the
+51% attacker's fork is obliged to _also_ process the _same_ sBTC transfers in
+order to keep their fork canonical.
+
+# Related Work
+
+This section is not done yet.  Notes follow.
+
+* PowPeg in RSK
+
+	* Requires HSMs on a majority of Bitcoin miners (brittle;
+closed-membership)
+
+	* Has a token with coinbases to incentivize validator
+liveness regardless of tx volume (good!)
+
+	* Probably the closest system to what
+we're building
+
+* Sidechains / drivechains
+
+	* No incentive for validators to
+process transactions, outside of miniscule tx fees
+
+	* No compelling answer for
+dealing with sidechain reorgs (Bitcoin miners have to fully validate, which
+does not scale)
+
+* Liquid
+
+	* No incentive for validators to process transactions,
+outside of miniscule tx fees
+
+	* Closed-membership federation is brittle and its
+admission process is opaque
+
+	* Why should users trust certain federation
+members' decisions? 
+
+* Merged mining
+
+	* Requires majority Bitcoin miner
+
+participation (risky without 100% participation -- one Bitcoin miner could have
+majority merged-mining power, like with Namecoin)
+
+	* Bitcoin miners must
+validate merged-mined chain (does not scale)
+
+None of the above can recover from a peg failure.
```

Comment:
Merged-mining is orthogonal to BTC bridge protocols, no? I mean, the techniques can be combined (as with Rootstock, which is both merge-mined and has a BTC bridge) but on its own merged-mining is just a block production Sybil-resistance mechanism that piggy-backs off the mining power of an existing blockchain -- closer in comparison to PoX than it is to sBTC.

Reply:
Right -- this bulleted collection speaks more to the limits of merged mining as a useful protocol for building a BTC bridge.  This will be clarified when this section is fully written.